OctoViewer/Coordinator.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  Coordinator.swift
+//  OctoViewer
+//
+//  Created by Hesham Salman on 5/26/17.
+//  Copyright © 2017 Hesham Salman
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+import UIKit
+
+class Coordinator {
```

Comment:
So I'm exploring an MVVMF pattern.

The idea is this: 
- Model (self explanatory)
- View (Views & View Controller). Responsible for visual presentation, but contain _no_ business logic. 
- View Model handles all of the business logic. 
- Flow **coordinator**: handles all navigation in the application. 

The coordinators in there at present aren't sufficient for scaling, because the base `Coordinator` is a class instead of a protocol, and inexplicably, it forces an optional navigation controller. It's going to take some thought for how to best structure this but I think it's best to move navigation to its own construct.